nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -307,142 +287,128 @@ def calculate_metrics(self, data: Dict) -> Dict[str, Tensor]:
 
 class SparsityAllocator:
     """
-    An abstract class for allocate mask based on metrics.
+    A base class for allocating mask based on metrics.
 
     Parameters
     ----------
     pruner
         The pruner that binded with this `SparsityAllocator`.
-    dim
-        The under pruning weight dimensions, which metric size should equal to the under pruning weight size on these dimensions.
-        None means one-to-one correspondence between pruned dimensions and metric, which equal to set `dim` as all under pruning weight dimensions.
-        The mask will expand to the weight size depend on `dim`.
-
-        Example:
-
-        The under pruning weight has size (2, 3, 4), and `dim=1` means the under pruning weight dimension is 1.
-        Then the metric should have a size (3,), i.e., `metric=[0.9, 0.1, 0.8]`.
-        Assuming by some kind of `SparsityAllocator` get the mask on weight dimension 1 `mask=[1, 0, 1]`,
-        then the dimension mask will expand to the final mask `[[[1, 1, 1, 1], [0, 0, 0, 0], [1, 1, 1, 1]], [[1, 1, 1, 1], [0, 0, 0, 0], [1, 1, 1, 1]]]`.
-    block_sparse_size
-        This used to describe the block size a metric value represented. By default, None means the block size is ones(len(dim)).
-        Make sure len(dim) == len(block_sparse_size), and the block_sparse_size dimension position is corresponding to dim.
-
-        Example:
-
-        The metric size is (12,), and block_sparse_size=[64], then the mask will expand to (768,) at first before expand with `dim`.
+    scalors
+        Scalor is used to scale the masks' size. It shrinks the mask of the same size as the pruning target to the same size as the metric,
```

Comment:
how about: It groups a bunch of values in target tensor to calculate one metric value for the group.

what confuses me the most is that what "mask size" represents. the same as target tensor size? the size after shrinking?